node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs

```diff
@@ -59,531 +41,195 @@ const CONTRACT_ABI: &str = indoc!(
-                            if transactions.is_empty() && logs_len != transactions.len() {
-                                warning!(
-                                    logger,
-                                    "Retrieving transactions: logs: {}, transactions: {}",
-                                    logs_len,
-                                    transactions.len()
-                                )
-                            }
-                            // Get the largest transaction block number, unless there are no
-                            // transactions, in which case use end_block, unless get_latest_block()
-                            // was not successful.
-                            let transaction_max_block_number = self
-                                .find_largest_transaction_block_number(
-                                    response_block_number,
-                                    &transactions,
-                                );
-                            debug!(
-                                logger,
-                                "Discovered transaction max block nbr: {}",
-                                transaction_max_block_number
-                            );
-                            Ok(RetrievedBlockchainTransactions {
-                                new_start_block: 1u64 + transaction_max_block_number,
-                                transactions,
-                            })
-                        }
-                    }
-                    Err(e) => {
-                        error!(self.logger, "Retrieving transactions: {:?}", e);
-                        Err(BlockchainError::QueryFailed(e.to_string()))
-                    }
-                }
-            }
-            Err(e) => Err(BlockchainError::QueryFailed(e.to_string())),
-        }
+                debug!(
+                    logger,
+                    "Retrieving transactions from start block: {:?} to end block: {:?} for: {} chain_id: {} contract: {:#x}",
+                    start_block,
+                    response_block_number,
+                    recipient,
+                    num_chain_id,
+                    contract_address
+                );
+                let filter = FilterBuilder::default()
+                    .address(vec![contract_address])
+                    .from_block(start_block)
+                    .to_block(BlockNumber::Number(U64::from(response_block_number)))
+                    .topics(
+                        Some(vec![TRANSACTION_LITERAL]),
+                        None,
+                        Some(vec![recipient.into()]),
+                        None,
+                    )
+                    .build();
+                lower_level_interface.get_transaction_logs(filter)
+                    .then(move |logs| {
+                        debug!(logger, "Transaction retrieval completed: {:?}", logs);
```

Comment:
First, it's a little lie that "transactions" were retrieved. What you are holding at the moments are just the Ethereum logs. But I don't think we'd want to print the whole logs. They are quite big. Reasonably bigger than the actual transaction which we want to make out of these. 

So what should we do? I advise to change the word Transactions for Logs and also to use trace! instead of debug! (because the dump of the logs will be messy and we don't really care about all the information in them).